server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -112,6 +112,8 @@
 
     private Map<String, String> dynamicTemplates = Map.of();
 
+    private Object rawTimestamp;
```

Comment:
hi, @martijnvg , will the rawTimestamp be cleaned up after a new source set? If so, it will cause the setRawTimestamp and getRawTimestamp not the same, when a new source set. It's a little strange.

Reply:
No, it will not be cleaned up. I don't think a new source will be set after `setRawTimestamp(...)` is invoked.
I think to be sure we can maybe add an assertion to `IndexRequest#source(...)` setter and if `rawTimestamp` is not null then fail.

The `rawTimestamp` field is only needed on coordinating node when resolving the data stream defined in `index`  field to a backing index.